collections/ipnskeys.go

```diff
@@ -65,6 +68,29 @@ func (k *IPNSKeys) GetByCid(ctx context.Context, cid string) (*IPNSKey, error) {
 	return decodeIPNSKey(raw)
 }
 
+func (k *IPNSKeys) ListByThreadID(ctx context.Context, threadID thread.ID) ([]IPNSKey, error) {
+	cursor, err := k.col.Find(ctx, bson.M{"thread_id": threadID.Bytes()})
```

Comment:
It seems that `cursor` has a Close function.
Should this be deferred?
Might apply to more places if that's the case.

Reply:
Ah yeah, good eye. Fixing.